src/server/services/RepositoryService.ts

```diff
@@ -151,7 +151,7 @@ export class RepositoryService implements IRepositoryService {
   ): Array<FileObject> {
     const fullPath: string =
       this.repositoryPath +
-      `/contracts/${match}/${chain}/${toChecksumAddress(address)}/`;
+      `/contracts/${match}/${chain}/${getAddress(address)}/`;
```

Comment:
this comment is valid for all the `getAddress`: there is a weird behavior, if the address has both uppercase and lowercase, it is considered formatted, but if it is wrongly formatted it throws an error (see the exampl here: https://docs.ethers.org/v5/api/utils/address/#utils-getAddress). IDK if we want to handle this

Reply:
Hmm isn't that actually a good behavior? I don't expect any mixed cased addresses, and in fact, such an address is invalid. 

Also bear in mind to check v6 docs and not v5 https://docs.ethers.org/v6/api/address/#getAddress

Reply:
I'm ok with using the ethers behavior, I'm just saying it will be different from before. For me we can keep it like you implemented it.